src/components/search-bar/search-bar.jsx

```diff
@@ -16,14 +18,34 @@ import './search-bar.css';
 export default class SearchBar extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
+      query: {
+        url: null,
+        startDate: null,
+        endDate: null
+      }
+    };
     this._didSearch = this._didSearch.bind(this);
     this._dispatchSearch = debounce(this._dispatchSearch.bind(this), 500);
+    this.urlSearch = this.urlSearch.bind(this);
+    this.dateSearch = this.dateSearch.bind(this);
```

Comment:
I’m OK if you don’t want to use `_` to prefix private methods, but we should stick to one style within this component. :)